### PR TITLE
PWX-31805: Container ochestrator support

### PR DESCRIPTION
**Pull request by jrivera-px**
**What this PR does / why we need it**:
The container orchestrator is a PX component which saves/retrieves/deletes the telemetry registration certificate from the k8s secrets.    The original ccm-go handled the registration certificate management and stored it directly into the k8s secrets on behalf of PX.   However the new ccm-go removes that special PX secret handling and uses a GRPC api to access a server in PX which will handle the certificate secret management.

This PR provides the spec changes necessary to give the new ccm-go the API connection information necessary to access the PX container orchestrator server. 
 
**Which issue(s) this PR fixes** (optional)
Closes #
PWX-31805

**Special notes for your reviewer**:
Testing spreadsheet provided: https://portworx.atlassian.net/browse/PWX-31767
This change is dependent on PX PR:   https://github.com/portworx/porx/pull/12890
Integration test PR:  https://github.com/portworx/torpedo/pull/2016

### CODE REVIEW

File: drivers/storage/portworx/component/telemetry.go [hunk 1]
```diff
@@ -1237,6 +1251,14 @@ func GetCCMListeningPort(cluster *corev1.StorageCluster) int {
 	return startPort + 20
 }
 
+func GetPxCOListeningPort(cluster *corev1.StorageCluster) int {
+	startPort := pxutil.StartPort(cluster)
+	if startPort == pxutil.DefaultStartPort {
+		return pxDefaultCOListeningPort
+	}
+	return startPort + 21
```
Comment (dahuang-purestorage) [inline drivers/storage/portworx/component/telemetry.go:1259]: Is +21 the default port for PXCOListening Port? 
If so, can we make it a variable?

File: drivers/storage/portworx/component/telemetry.go [hunk 2]
```diff
@@ -581,6 +588,13 @@ func (t *telemetry) createCCMGoConfigMapRegister(
 	if err != nil {
 		return err
 	}
+
+	// Check PX version to update the config for "container orchestrator"
+	if pxutil.IsCOSupported(cluster) {
+		config = certStoreTypeRegx.ReplaceAllString(config,
+			"certStoreType: \"kvstore\"\nkvStore:\n  serverAddress: \"127.0.0.1:"+strconv.Itoa(GetPxCOListeningPort(cluster))+"\"")
```
Comment (dahuang-purestorage) [inline drivers/storage/portworx/component/telemetry.go:595]: not sure if i'm readying this right. But should we add a new line at the end too?

Comment (jrivera-px) [reply drivers/storage/portworx/component/telemetry.go:595]: The configmap by defaults contains 
```
cloud:
      auth:
        version: "1.0"
    productName: "portworx"
    certStoreType: "k8s"
    k8s:
      certSecretName: "pure-telemetry-certs"
      certSecretNamespace: kube-test
``` 
after the replace code it already has a newline
```
 cloud:
      auth:
        version: "1.0"
    productName: "portworx"
    certStoreType: "kvstore"
    kvStore:
      serverAddress: "127.0.0.1:9030"
    k8s:
      certSecretName: "pure-telemetry-certs"
      certSecretNamespace: portworx
```

File: drivers/storage/portworx/component/telemetry.go (hunk 1 above)
Comment (jrivera-px) [reply drivers/storage/portworx/component/telemetry.go:1259]: Sure let me make that change...

**Review by dahuang-purestorage — APPROVED** [review]
LGTM

**Pull request merged**